packages/strapi-helper-plugin/lib/internals/webpack/webpack.base.babel.js

```diff
@@ -2,153 +2,52 @@
-      if (_.get(server, 'admin.build.host')) {
-        URLs.host = _.get(server, 'admin.build.host', '/admin').replace(/\/$/, '') || '/';
-      } else {
-        URLs.host = _.get(server, 'admin.path', '/admin');
-      }
-
-      URLs.publicPath = URLs.host;
-      URLs.backend = _.get(server, 'admin.build.backend', '/');
-
-      if (_.get(server, 'admin.build.plugins.source') === 'backend') {
-        URLs.mode = 'backend';
-      }
-
-      if (process.env.npm_lifecycle_event === 'start') {
-        URLs.backend = `http://${_.get(server, 'host', 'localhost')}:${_.get(server, 'port', 1337)}`;
-      }
-    }
-  } catch (e) {
-    throw new Error(`Impossible to access to ${serverConfig}`);
-  }
-}
-
-// Load plugins into the same build in development mode.
-const plugins = {
-  exist: false,
-  src: [],
-  folders: {},
-};
-
-if (process.env.npm_lifecycle_event === 'start') {
-  try {
-    fs.accessSync(path.resolve(appPath, 'plugins'), fs.constants.R_OK);
-  } catch (e) {
-    // Allow app without plugins.
-    plugins.exist = true;
-  }
-
-  // Read `plugins` directory and check if the plugin comes with an UI (it has an App container).
-  // If we don't do this check webpack expects the plugin to have a containers/App/reducer.js to create
-  // the plugin's store (redux).
-  plugins.src =
-    isAdmin && !plugins.exist
-      ? fs.readdirSync(path.resolve(appPath, 'plugins')).filter(x => {
-          let hasAdminFolder;
-
-          try {
-            fs.accessSync(path.resolve(appPath, 'plugins', x, 'admin', 'src', 'containers', 'App'));
-            hasAdminFolder = true;
-          } catch (err) {
-            hasAdminFolder = false;
-          }
-          return x[0] !== '.' && hasAdminFolder;
-        })
-      : [];
-
-  // Construct object of plugin' paths.
-  plugins.folders = plugins.src.reduce((acc, current) => {
-    acc[current] = path.resolve(
-      appPath,
-      'plugins',
-      current,
-      'node_modules',
-      'strapi-helper-plugin',
-      'lib',
-      'src',
-    );
-
-    return acc;
-  }, {});
-}
-
-// Tell webpack to use a loader only for those files
-const foldersToInclude = [path.join(adminPath, 'admin', 'src')]
-  .concat(
-    plugins.src.reduce((acc, current) => {
-      acc.push(path.resolve(appPath, 'plugins', current, 'admin', 'src'), plugins.folders[current]);
-
-      return acc;
-    }, []),
-  )
-  .concat([path.join(adminPath, 'node_modules', 'strapi-helper-plugin', 'lib', 'src')]);
+const { COMMON_ALIAS } = require('./configs/alias');
+const foldersToInclude = require('./configs/foldersToInclude');
+const { __NODE_ENV__, __PROD__ } = require('./configs/globals');
+const paths = require('./configs/paths');
+const URLs = require('./configs/server');
+const pkg = require(paths.packageJson);
+const pluginId = pkg.name.replace(/^strapi-/i, '');
 
 module.exports = options => {
+  const { alias, babelPresets, devtool, disableExtractTextPlugin, entry, externals,  output, plugins } = options;
```

Comment:
I think the object destructuring makes the configuration harder to understand here.

Reply:
If you have a lot of code, it is meaningful to keep the grouping/object. Agreed